### DISCUSSION TITLE
Option to rename tracks

### DISCUSSION THREAD

**Discussion by RogueOneEcho** [RogueOneEcho/caesura#58] [Ideas]
- [ ] The logic to rename tracks following a pattern is already written and could easily be implemented under an optional `--rename-tracks` flag. 

Is this feature desired?

https://github.com/RogueOneEcho/caesura/blob/ae178eeb9ac4a1e0aff131dd5e26c0980fc175c3/src/naming/track_name.rs#L1-L15

- [ ] Multi disks would need to be written to different folders 
- [ ] or prefix the track with disk number? ie. Disk 2 Track: `02-03` or `203`

This would allow transcoding of sources which are trumpable due to bad file/folder structure

**Comment by windexvalence** [comment]
This would be very helpful for sure.

**Comment by damnmersault** [comment]
Being able to force the directory name would also be helpful. For example, with classical albums the standard naming convention can easily exceed 180 characters, and so it's common to drop the artist name from the top level folder and start with just the album name.

**Reply by RogueOneEcho** [reply]
I've created a separate idea to track this: https://github.com/RogueOneEcho/caesura/discussions/155

**Comment by RogueOneEcho** [comment]
Working prototype is now complete: https://github.com/RogueOneEcho/caesura/tree/feat-rename-tracks 

I need to do some thorough checking before it gets merged into main and released.

**Reply by RogueOneEcho** [reply]
Completed in 57f5c235
